Sources/PIALibrary/Account/Domain/UseCases/APITokenUseCase.swift

```diff
@@ -0,0 +1,41 @@
+
+import Foundation
+
+public protocol APITokenUseCaseType {
+    typealias Completion = (() -> Void)
+    func getAPIToken() -> APIToken?
+    func refreshAPIToken(completion: APITokenUseCaseType.Completion)
+}
+
+class APITokenUseCase: APITokenUseCaseType {
+    
+    private let apiTokenKey = "API_TOKEN_KEY"
+    
+    let keychainStore: SecureStore
+    let tokenSerializer: AuthTokenSerializerType
+    
+    init(keychainStore: SecureStore, tokenSerializer: AuthTokenSerializerType) {
+        self.keychainStore = keychainStore
+        self.tokenSerializer = tokenSerializer
+    }
+    
+    
+    public func getAPIToken() -> APIToken? {
+        guard let tokenDataString = keychainStore.token(for: apiTokenKey),
+              let tokenData = tokenDataString.data(using: .utf8) else { return nil }
+        return tokenSerializer.decodeAPIToken(from: tokenData)
+    }
+    
+    public func refreshAPIToken(completion: () -> Void) {
```

Comment:
Just out of curiosity, why using completion closure instead of async functions? Same in `VPNTokenUseCase`

Reply:
For iOS 12 compatibility. I am actually planning to expose 2 APIs: 
- One with async await for iOS 13+
- This one with the completion handler for iOS 12

Reply:
Right, completely forgot about that.
Awesome, that you @kp-laura-sempere!